ci-operator/step-registry/rosa/cluster/index/rosa-cluster-index-commands.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+function is_valid_json(){
+  file=$1
+  if [ ! -f "${file}" ]; then
+    echo "ERROR: File ${file} not found"
+    return 1
+  elif jq '.' "${file}" >/dev/null 2>&1; then
+    echo "INFO: File ${file} contains a valid json"
+    return 0
+  else
+    echo "ERROR: File ${file} is not a valid json"
+    return 1
+  fi
+}
+
+function is_valid_es(){
+  URL="${1}"
+  INDEX="${2}"
+  if [[ $(curl -sS "${URL}" | jq -r .cluster_name) == "null" ]]; then
+    echo "ERROR: Cannot connect to ES ${URL}"
+    return 1
+  elif [[ $(curl -sS "${URL}"/_cat/indices | awk -v index_name="${INDEX}" '$3 == index_name {print $2}') != "open" ]]; then
+    echo "ERROR: Index ${URL}/${INDEX} not healthy"
+    return 1
+  else
+    echo "INFO: ${URL}/${INDEX} healthy"
+    return 0
+  fi
+}
+
+index_metadata(){
+  METADATA="${1}"
+  URL="${2}/${3}/_doc"
+  cat "${METADATA}"
+  RESULT=$(curl -X POST "${URL}" -H 'Content-Type: application/json' -d "$(cat ${METADATA})" 2>/dev/null)
+  if [[ $(echo "${RESULT}" | jq -r .result) == "created" ]]; then
+    echo "INFO: Index of ${METADATA} completed"
+    return 0
+  else
+    echo "ERROR: Failed to index ${METADATA}"
+    echo "${RESULT}"
+    return 1
+  fi
+}
+
+ES_SERVER=$(cat "/secret/host")
+
+if is_valid_json "${SHARED_DIR}/${METADATA_FILE}" && is_valid_es "${ES_SERVER}" "${ES_INDEX}" ; then
```

Comment:
if there is no ES_SERVER defined, index wont be executed, so any team without ES can execute the job without failures

Reply:
did we talked about changing anything here? I dont remember. indexing will be executed only when there will be an ES_SERVER var, which is added only on perfscale workflows, so it following the comment from @yasun1 
@mukrishn @yasun1

Reply:
Yes, I think we did and fine with implementation with the fact that it will be called only with respective variable is set. @yasun1 please correct us if we are missed to capture anything in this part.

Reply:
added a new var, INDEX_ENABLED, set to false by default, and added as true only on openshift-qe-ocp-qe-perfscale-ci-main__rosa jobs

this will be the entry point of the index script, if not set to "true", anything else wont be executed